apps/nextra/pages/en/build/smart-contracts/linter.mdx

```diff
@@ -89,6 +98,17 @@ Check for boolean expressions that can be simplified when a boolean literal (eit
 Does NOT consider side-effects/short-circuiting in the recommended simplifications. Example:
 - `1/0 || true` is logically equivalent to `true`, but applying this simplification affects program semantics.
 
+### `self_assignment`
+
+Checks for patterns where a variable or a field of a struct is assigned to itself and suggests removing the assignment. These assignments do not affect the state of the program. Examples include:
+- `let x = x;`
+- `x = x;`
+- `a.x = a.x;`
+
+Note that this does not detect patterns involving non-builtin functions, including vector operations, nor does this detect patterns that deal with global storage. For example, the following are out of scope of this lint check:
```

Comment:
I think we can say above "checks for some simple patterns where .." and leave out listing what is out of scope.